src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1884,13 +1883,15 @@ void ImportAddressOfField(int token, bool isStatic)
             bool isPermanentHome = false;
 
             TypeDesc instance;
-
             if (isStatic)
             {
                 Check(field.IsStatic, VerifierError.ExpectedStaticField);
 
                 isPermanentHome = true;
                 instance = null;
+
+                if (field.IsInitOnly)
+                    Check(_method.IsStaticConstructor && field.OwningType == _method.OwningType, VerifierError.Initonly);
```

Comment:
Nit - change the casing of VerifierError.Initonly to VerifierError.InitOnly to match the property name?